deepmd/pt/infer/deep_eval.py

```
@@ -513,6 +531,49 @@ def _eval_model(
             )
         else:
             aparam_input = None
+        if atomic_weight is not None:
+            # Validate atomic_weight shape
+            if atomic_weight.ndim == 1:
+                # (natoms,) - broadcast across frames
+                if atomic_weight.shape[0] != natoms:
+                    raise ValueError(
+                        f"atomic_weight shape {atomic_weight.shape} incompatible with number of atoms {natoms}"
+                    )
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)